### ISSUE TITLE
Custom UDF Fields

### ISSUE THREAD

**Issue by Vinsanity** [skyverge/woocommerce-sage-erp-connector#11]
Quick question if someone can help me out. 
I'm having some trouble getting data over to custom UDF fields within SAGE. I've tried mapping this data with `$sales_order->UDF_FIELD_NAME`, `$sales_order->OtherFields->UDF_FIELD_NAME`, `$customer->UDF_FIELD_NAME`, and `$customer->OtherFields->UDF_FIELD_NAME`. 

Neither have worked for populating data in a SAGE order. I checked the response from the API and the output of available fields shows the OtherFields object but no custom fields are showing up there even though they're been set to be exposed to the API.

Any insight or help would be greatly appreciated.

Thanks

**Comment by maxrice** [comment]
try `$sales_order->_CustomFields = array( 'UDF_DELIVERY_NOTE$' => 'some note' );`

**Comment by Vinsanity** [comment]
Hi @maxrice 
Thanks for the response. I see what you mean with using `_CustomFields`. Was able to trace that back and found the private methods that unpack and pack them for SAGE usage. 

I have it set like this now:

```
$sales_order->_CustomFields = array(
'UDF_DELIVERY_NOTE$' => $value_1,
'UDF_DELIVERY_CONTACT$' => $value_2,
);
```

I'm still running into issues however with this but it seems to be related to not receiving an exception error. Sounds like it's on my client's end. 

error reads: 

> Export Failure - Order 26851 (ERP Sales Order No: 0044808), Message : The server was unable to process the request due to an internal error. For more information about the error, either turn on IncludeExceptionDetailInFaults (either from ServiceBehaviorAttribute or from the configuration behavior) on the server in order to send the exception information back to the client, or turn on tracing as per the Microsoft .NET Framework SDK documentation and inspect the server trace logs. (N/A)

If/when I get this cleared up I'll let you know.

**Comment by Vinsanity** [comment]
Hello @maxrice ,
I'm having a bit of trouble getting some info from the plugin. What you recommend is the easiest way to test/see the output of the $sales_order object?

Also, is this being sent to SAGE as XML or is it through JSON/something else.

Thanks

**Comment by maxrice** [comment]
i'd use the filter and `var_dump()` the object.

The sage API is SOAP, so the data is sent as XML under the WSDL schema Sage defines.

**Comment by Vinsanity** [comment]
Thanks @maxrice ,

I'm using the `wc_sage_erp_connector_sales_order` filter and doing `var_dump($sales_order)` however I can't seem to get this to work. I've tried this both outside my extension of the master plugin and by adding `var_dump` to my extension at the end of function before returning the $sales_order 

`add_filter( 'wc_sage_erp_connector_sales_order', array( $this, 'modify_sales_order' ), 10, 2 );`

```
public function modify_sales_order( $sales_order, $order ) {
... //my modification code up here
var_dump( $sales_order );
return $sales_order;
}
```

I also tried capturing this in dev tools ajax response but no luck there. The weird thing is I am getting the orders over to SAGE successfully but cannot get the response to output.

Any help is greatly appreciated.
Thanks

**Comment by maxrice** [comment]
yeah this output is probably lost during execution, you'll want to log it to the PHP error log, using something like:

```php
ob_start();
var_dump( $sales_order );
error_log( ob_get_clean() );
```
